File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit =0;
        int n = prices.length;
        int buy = 0 ,sell = 1;

        while (sell < n){
            if(prices[sell] > prices[buy]){
                profit = Math.max(profit,prices[sell]-prices[buy]);
            }
            else{
                buy=sell;
            }
            sell++;
        }

        return profit;
        
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n = len(prices)
        buy =0
        sell = 1
        profit = 0

        while sell < n:
            if prices[sell] > prices[buy]:
                profit = max(profit,prices[sell]-prices[buy])
            else:
                buy=sell
            sell+=1
        
        return profit